Controller/Adminhtml/Export/Download.php

```diff
@@ -0,0 +1,68 @@
+<?php declare(strict_types = 1);
+
+namespace Hyva\Admin\Controller\Adminhtml\Export;
+
+use Hyva\Admin\Model\Export;
+use Hyva\Admin\ViewModel\HyvaGridInterface;
+use Magento\Backend\App\Action;
+use Magento\Backend\App\Action\Context;
+use Magento\Framework\App\Action\HttpGetActionInterface;
+use Magento\Framework\App\RequestInterface;
+use Magento\Framework\App\ResponseInterface;
+use Magento\Framework\App\Response\Http\FileFactory;
+
+class Download extends Action implements HttpGetActionInterface
+{
+    const GRID_NAME = 'gridName';
+
+    protected HyvaGridInterface $grid;
+
+    protected Export $export;
+
+    private RequestInterface $request;
+
+    private FileFactory $fileFactory;
+
+    private ResponseInterface $response;
+
+    public function __construct(
+        Context $context,
+        RequestInterface $request,
+        ResponseInterface $response,
+        FileFactory $fileFactory,
+        Export $export
+    ) {
+        parent::__construct($context);
+        $this->request = $request;
+        $this->fileFactory = $fileFactory;
+        $this->response = $response;
+        $this->export = $export;
+    }
+
+    public function execute()
+    {
+        $export = $this->export->getExport(
+            $this->request->getParam(self::GRID_NAME, ''),
+            $this->request->getParam('exportType', '')
+        );
+        $this->prepareRequest();
+        $export->create();
+        $this->response = $this->fileFactory->create(
+            basename($export->getFileName()),
+            [
+                "type"  => "filename",
+                "value" => $export->getFileName(),
+                "rm"    => true,
+            ],
+            $export->getRootDir(),
+            $export->getMetaType()
+        );
+        $this->response->sendResponse();
+    }
+
+    private function prepareRequest()
+    {
+        $params = array_diff_key($this->request->getParams(), array_flip(['p', 'key', 'exportType', 'ajax']));
+        $this->request->clearParams()->setParams($params);
```

Comment:
I've changed the name so far and I agree that the button should have the right parameters for the export.
But the filtering is done with ajax in the filter component and the request parameters are bound to the form, so that every change in the form (e.g. filter by name) will also get passed to the button and the controller. 

The only params respected here are the paging, formkey, exportType (which is internally) and ajax - so I think the controller doesn't really know what is going on somewhere - but I really appreceate your input on that. 
Do you have any better recommendations?